File: module_main/src/main/java/com/yyxnb/module_main/ui/SplashActivity.java
package com.yyxnb.module_main.ui;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.yyxnb.lib_arch.base.IActivity;
import com.yyxnb.module_main.R;

/**
 * 启动页
 *
 * @author yyx
 */
public class SplashActivity extends AppCompatActivity implements IActivity {

    @Override
    public int initLayoutResId() {
        return R.layout.activity_splash;
    }

    @Override
    public void initView(@Nullable Bundle savedInstanceState) {
        // 白屏时间 + 2秒进入首页 ≈ 3~5秒
        handler.sendEmptyMessageDelayed(1, 2 * 1000);
    }

    private final Handler handler = new Handler(message -> {
        if (message.what == 1) {
            inMain();
        }
        return false;
    });

    private void inMain() {
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        // 避免内存泄漏
        handler.removeCallbacksAndMessages(null);
    }
}

File: module_main/src/main/java/com/yyxnb/module_main/ui/SplashActivity.kt
package com.yyxnb.module_main.ui

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Message
import androidx.appcompat.app.AppCompatActivity
import com.yyxnb.arch.base.IActivity
import com.yyxnb.module_main.R

/**
 * 启动页
 *
 * @author yyx
 */
class SplashActivity : AppCompatActivity(), IActivity {

    override fun initLayoutResId(): Int {
        return R.layout.activity_splash
    }

    override fun initView(savedInstanceState: Bundle?) {
        // 白屏时间 + 2秒进入首页 ≈ 3~5秒
        handler.sendEmptyMessageDelayed(1, 2 * 1000.toLong())
    }

    private val handler = Handler { message: Message ->
        if (message.what == 1) {
            inMain()
        }
        false
    }

    private fun inMain() {
        startActivity(Intent(this, MainActivity::class.java))
        finish()
    }

    override fun onDestroy() {
        super.onDestroy()
        // 避免内存泄漏
        handler.removeCallbacksAndMessages(null)
    }
}